draft-ietf-ppm-dap.md

```diff
@@ -1307,12 +1365,13 @@ struct {
 } AggregationJobResp;
 ~~~
 
-The message is a sequence of PrepareStep values, the order of which matches that
-of the ReportShare values in `AggregationJobInitReq.report_shares`. Each report
-that was marked as invalid is assigned the `PrepareStepState` `failed`.
+The message is a sequence of `PrepareStep` values, the order of which matches
+that of the ReportShare values in `AggregationJobInitReq.report_shares`. Each
+report that was marked as invalid is assigned the `PrepareStepState` `failed`.
 Otherwise, the `PrepareStep` is either marked as continued with the output
-`prep_msg`, or is marked as finished if the VDAF preparation process is finished
-for the report share. The Helper's aggregation job is now in round 0.
+`prep_msg` & `prep_share`, or is marked as finished if the VDAF preparation
+process is finished for the report share with the final `prep_msg`. The Helper's
+aggregation job is now in round 0.
```

Comment:
I get the distinction you're drawing between DAP round and VDAF round, and agree that for a 1-round VDAF, it won't really matter, because on retry, the leader will re-send `AggregationJobInitReq`. But then I think we need to make sure the leader and helper continuation sections make it clear when either advances to the next round. Right now, the discussion of how the leader constructs `AggregationJobContinueReq` reads (lines 1596-1597):

>The `round` field is the round of **VDAF preparation** that the Leader just reached
>and wants the Helper to advance to.

Emphasis mine. I think it might suffice to change that from "VDAF preparation" to something like "DAP aggregation".

I put together a pair of diagrams to work out just when the DAP and VDAF round transitions occur to make sure I understand this. Would you take a look and confirm to me that we're understanding this the same way?

[2 round VDAF, no failures](https://sequencediagram.org/index.html#initialData=CoSwLgNgpgBATDATgewK4DsAmMBqARAQQDEYAKKAOgHMKYAFZABwgENEBKAKE8bbBADGIXujAwAMlBaYoiHn0HCWomAAkoERrO4AjZAA8YyAG6yJUmYgBc0aWYDOYFmFgAGGAF5cmFgDMKjIhQjAD6IOjgpLaWSMHIiGL2ABZsUOwAOuhRFg5OLjAAjAA0MNFmgcEwyakwruye3n4BQaHoUPpg2XaIVXluJQByyG1ceoYmZpLdVo7xsCxUVEFUziDDMABWyDowAO7gSUhoWA2uJWU9js6wBZyIY0amPVOWVlCiZvjERxjYrpnvFw9Qh0H4nVycF6yAC0AD51JpZFYCItlqthgApbYASQiYAASlAAI6kJIaLQ9FrxRIpILnHI9CqMKq0tyjAyPMwIilWMmIy59WoNHA+fxMsJ40nksxUhIs1IZLJ8im9a6FErK8oteVBWr1LwipritodKX81X5M4wIYjO4PCY9blIwGfQgkFC-Wqce1PNTS6xMmAAW3sVCFBtFzWCIWqQXsITAyBCTLNKsDsf6pQZMHTrL1mVTuTVcA1-pz2ozhX1jTFLRCJs6moFauK5cqIbDdW9HIdfv5M0TuoWSygK346y2O32YEOHpOXitTYtsDgdp7vqd1hdPS+7uO2AKAI+wIIoLnf04m7hUOsKJHY7W6CxOkJ9kYpDoLVSAGUXIwrFYADCwz8OgqBQKQgYdrUpbmrmNQFOw7LjL6N5WL4UBgAIhzDmi47oJs2x7AcYLYAu9LdMuhTdihkwMlYXQxFc+QllmlHwbqiHCpGxrtJ0FxUa2UGhnqa60c89HbrgbqkdRPp0dMwlhgU3FGnWGbxomyYtIxWqVJWrZLhxNyKmgNI1BGanRg2unNixJRKVWNGchJ0y9HMMC4aO6IEZOxEzrJXitmZOpQGJLnmNMUm7rJcBHkCMAgrJtw3nCm7Iqi3n4c+wGiOE4GEiS54NK2n7BD+f4AUQ4QgMkUCYJB2rQYhyERelGFYThmUPhORHTrO+6nLBKrMSuzm9ulIWVpZtbWXxhZ2SuDlNSJiHjRu-oDh5Xk9b5fUkcVQUlFNrLhRNm3RTJxVxW0x6JaeyWXv6170XeeGPs+r7vmVvBBL+wRVTVdUNQA3gAvuwQA)

[2 round VDAF, response dropped during continuation](https://sequencediagram.org/index.html#initialData=CoSwLgNgpgBATDATgewK4DsAmMBqARAQQDEYAKKAOgHMKYAFZABwgENEBKGAd3AAsYAZixARUiKACgJjNmBABjEDPRgYAGSgtMURNNkKlLFTAASUCIx1SARsgAeMZADcd6zdsQAuaFtcBnMBYwWAAGGABeXEwWAQpGcUYAfRB0cFIfDyQoRmREVT9eNih2AB10dPd-QOCYAEYAGhgM13jsmAKimBDOSJxo2Nak9Cg7MArfRHbq0MaAOWRh9glbB2dXDQnPANzYFioqcSogkAWYACtka24+JDQsCK7G5smAoNhaiUQVxxdJjY9PFAVK58MRbhhsCEykDgpNCHRwfcQhJ-joALQAPjMFh0ngI+0OxwWAClLgBJVJgABKUAAjqReOZLJMErl8oVxE9KpNBu0OaElt81pNsczPIycS9pl0Hn0YnEEslKQyma5WXk+UVSuUJcypm86o1dS0EprxF0elF5YNEsNRirJfqaiE5gtip8hb9TKqvDCQYQSCgIV1lvYfq5RbjeQBbPxUGW9foK7KJDriPyJMDIRKDB163lpmZNbkwAv8i1lPNVA1wI0+0umwt1S1ygaKu1jY1Sg0NBttWPx7qh1ZeyNebbmvYHKBHOSnC5XHhgfhB+6RF0wLtO2BwD1h4XeyWA4GTUGBu7YWrQk8weGIyESMeY1FefHT2cndCk6w0vyMUh0AkRQAMrBIwnieAAwgscjoKgUCkDGcaPJu9Zlp0tTsIK+5ei+ngCFAYDyPwU6EnO6DnJc1zLveDwbs8251MO4Z-NynjjJkrw1LWxYTH2jBmu8LZJjaHYcdWNS9khg5LF8OHrGxfqngGtEfJ6CmbNJdSyiJiqFhmWY5gk4k8o25a9lu6HmphZRoOynSJta7YjGMDFcTujRaZhzEHnhUw7DApEzkSFELtRK4Xg8vZ2YJe4jhpAJKbgKmrtgcDXrCt4EAiqVMS+mJjniBLBeR37QSoKTwTS9K5ZEvaAdkIFgRBRApCABRQJgiGmgOzbYfFIo+vhhHEYFxUfvOVFLhFwbrnWjrufAPmjkNMVNo5bYpmJW6LTxXn9Sxh5ihOuzjSFlGLjctWGo4qD2eIcWHYVSVnrR6XDDed65R8T4AMTiJgngAMx2Hhb5kZ+36-v+DUyOIoHZC1bUdV1ADeAC+B2+Wx4hgIgACeUzZIwKTxkFE2hZcKLcgVQ3gyVkOXOVsFVXSpDXfVQHw81nitakKPdf2yGYVjK1HgRREkWd5EXeFtF1Y0a38stEZDeT51he1gUQOIWiE6cX2Rb2CSsIT2jBPIwSYI+PrPmx9MU1DUB-gBXNQAj4G88jjJo5jEhAA)

Reply:
Good point -- I switched the description of the `round` field to explicitly describe DAP rounds rather than VDAF rounds.

I think the sequence diagrams are roughly correct. (I'm not sure what it means to "enter into" a round, or why the rounds are entered at the points where they are in the sequence, but the counts are correct vs network round trips.)

Ultimately I think it's more useful to count DAP rounds over VDAF rounds because the purpose of the `round` field is to enable retries. The unit of retries in DAP is the network request, rather than a VDAF round. (this is confounded by the fact that, before this change, a DAP network request could be 1-1 mapped to a VDAF round.)